library/X509/Common/Database.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/* Icinga Web 2 X.509 Module | (c) 2022 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\X509\Common;
+
+use Icinga\Data\ResourceFactory;
+use ipl\Sql;
+
+trait Database
+{
+    /**
+     * Get the connection to the X.509 database
+     *
+     * @return  Sql\Connection
+     */
+    protected function getDb()
+    {
+        $config = new Sql\Config(ResourceFactory::getResourceConfig(
+            $this->Config()->get('backend', 'resource')
+        ));
+
+        $options = [
+            \PDO::ATTR_DEFAULT_FETCH_MODE => \PDO::FETCH_OBJ
```

Comment:
Please `use PDO;` and what about `PDO::FETCH_BOTH`?

Reply:
> what about `PDO::FETCH_BOTH`?

Wouldn't that be a bit too much? `PDO::FETCH_ASSOC` is only used in the controller, elsewhere we only use `PDO::FETCH_OBJ`.

Reply:
> Wouldn't that be a bit too much?

Yes, maybe. Then please add `$options` to getDb() and unify them.